### ISSUE TITLE
K8s system stack not able to start with vxlan networking.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#6817]
Rancher-server version - v1.2.0-rc1

Steps to reproduce the problem:
Create a K8s template with vxlan enabled and ipsec disabled.
Create environment with this K8s template.
Add 3 hosts to the environment.

K8s system stack is not able to start successfully:

<img width="1106" alt="screen shot 2016-11-29 at 10 04 58 am" src="https://cloud.githubusercontent.com/assets/4266958/20722398/53121c24-b61b-11e6-9130-12651b3838ed.png">

Logs from kubernetes container:
```
11/29/2016 9:36:25 AMtime="2016-11-29T17:36:25Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:36:47 AMtime="2016-11-29T17:36:47Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:36:55 AMtime="2016-11-29T17:36:55Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:37:15 AMtime="2016-11-29T17:37:15Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:37:32 AMtime="2016-11-29T17:37:32Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:37:41 AMtime="2016-11-29T17:37:41Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:38:02 AMtime="2016-11-29T17:38:02Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:38:19 AMtime="2016-11-29T17:38:19Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:38:27 AMtime="2016-11-29T17:38:27Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:38:39 AMtime="2016-11-29T17:38:39Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:38:55 AMtime="2016-11-29T17:38:55Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:39:00 AMtime="2016-11-29T17:39:00Z" level=info msg="Adding Host k8svxlan1.localdomain 138.197.26.145"
11/29/2016 9:39:00 AMtime="2016-11-29T17:39:00Z" level=info msg="Adding Host k8svxlan2.localdomain 138.197.26.211"
11/29/2016 9:39:00 AMtime="2016-11-29T17:39:00Z" level=info msg="Adding Host k8svxlan3.localdomain 138.197.18.4"
11/29/2016 9:39:25 AMtime="2016-11-29T17:39:25Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:39:30 AMtime="2016-11-29T17:39:30Z" level=info msg="Adding Host k8svxlan1.localdomain 138.197.26.145"
11/29/2016 9:39:30 AMtime="2016-11-29T17:39:30Z" level=info msg="Adding Host k8svxlan2.localdomain 138.197.26.211"
11/29/2016 9:39:30 AMtime="2016-11-29T17:39:30Z" level=info msg="Adding Host k8svxlan3.localdomain 138.197.18.4"
11/29/2016 9:39:47 AMtime="2016-11-29T17:39:47Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:40:07 AMtime="2016-11-29T17:40:07Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:40:23 AMtime="2016-11-29T17:40:23Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:40:40 AMtime="2016-11-29T17:40:40Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:41:02 AMtime="2016-11-29T17:41:02Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:41:10 AMtime="2016-11-29T17:41:10Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:41:32 AMtime="2016-11-29T17:41:32Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:41:52 AMtime="2016-11-29T17:41:52Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:41:57 AMtime="2016-11-29T17:41:57Z" level=info msg="Adding Host k8svxlan1.localdomain 138.197.26.145"
11/29/2016 9:41:57 AMtime="2016-11-29T17:41:57Z" level=info msg="Adding Host k8svxlan2.localdomain 138.197.26.211"
11/29/2016 9:41:57 AMtime="2016-11-29T17:41:57Z" level=info msg="Adding Host k8svxlan3.localdomain 138.197.18.4"
11/29/2016 9:42:17 AMtime="2016-11-29T17:42:17Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:42:47 AMtime="2016-11-29T17:42:47Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:43:10 AMtime="2016-11-29T17:43:10Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:43:15 AMtime="2016-11-29T17:43:15Z" level=info msg="Adding Host k8svxlan1.localdomain 138.197.26.145"
11/29/2016 9:43:15 AMtime="2016-11-29T17:43:15Z" level=info msg="Adding Host k8svxlan2.localdomain 138.197.26.211"
11/29/2016 9:43:15 AMtime="2016-11-29T17:43:15Z" level=info msg="Adding Host k8svxlan3.localdomain 138.197.18.4"
11/29/2016 9:43:40 AMtime="2016-11-29T17:43:40Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:43:45 AMtime="2016-11-29T17:43:45Z" level=info msg="Adding Host k8svxlan1.localdomain 138.197.26.145"
11/29/2016 9:43:45 AMtime="2016-11-29T17:43:45Z" level=info msg="Adding Host k8svxlan2.localdomain 138.197.26.211"
11/29/2016 9:43:45 AMtime="2016-11-29T17:43:45Z" level=info msg="Adding Host k8svxlan3.localdomain 138.197.18.4"
11/29/2016 9:44:25 AMtime="2016-11-29T17:44:25Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:44:48 AMtime="2016-11-29T17:44:48Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:45:09 AMtime="2016-11-29T17:45:09Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:45:17 AMtime="2016-11-29T17:45:17Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:45:40 AMtime="2016-11-29T17:45:40Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:45:45 AMtime="2016-11-29T17:45:45Z" level=info msg="Adding Host k8svxlan1.localdomain 138.197.26.145"
11/29/2016 9:45:45 AMtime="2016-11-29T17:45:45Z" level=info msg="Adding Host k8svxlan2.localdomain 138.197.26.211"
11/29/2016 9:45:45 AMtime="2016-11-29T17:45:45Z" level=info msg="Adding Host k8svxlan3.localdomain 138.197.18.4"
11/29/2016 9:46:24 AMtime="2016-11-29T17:46:24Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:46:54 AMtime="2016-11-29T17:46:54Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:46:59 AMtime="2016-11-29T17:46:59Z" level=info msg="Adding Host k8svxlan1.localdomain 138.197.26.145"
11/29/2016 9:46:59 AMtime="2016-11-29T17:46:59Z" level=info msg="Adding Host k8svxlan2.localdomain 138.197.26.211"
11/29/2016 9:46:59 AMtime="2016-11-29T17:46:59Z" level=info msg="Adding Host k8svxlan3.localdomain 138.197.18.4"
11/29/2016 9:48:02 AMtime="2016-11-29T17:48:02Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:49:03 AMtime="2016-11-29T17:49:03Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:49:09 AMtime="2016-11-29T17:49:09Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:49:29 AMtime="2016-11-29T17:49:28Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:49:33 AMtime="2016-11-29T17:49:33Z" level=info msg="Adding Host k8svxlan1.localdomain 138.197.26.145"
11/29/2016 9:49:33 AMtime="2016-11-29T17:49:33Z" level=info msg="Adding Host k8svxlan2.localdomain 138.197.26.211"
11/29/2016 9:49:33 AMtime="2016-11-29T17:49:33Z" level=info msg="Adding Host k8svxlan3.localdomain 138.197.18.4"
11/29/2016 9:50:03 AMtime="2016-11-29T17:50:03Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:50:13 AMtime="2016-11-29T17:50:13Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:50:23 AMtime="2016-11-29T17:50:23Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:50:34 AMtime="2016-11-29T17:50:34Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:50:47 AMtime="2016-11-29T17:50:47Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:50:55 AMtime="2016-11-29T17:50:55Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:51:25 AMtime="2016-11-29T17:51:25Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:51:30 AMtime="2016-11-29T17:51:30Z" level=info msg="Adding Host k8svxlan1.localdomain 138.197.26.145"
11/29/2016 9:51:30 AMtime="2016-11-29T17:51:30Z" level=info msg="Adding Host k8svxlan2.localdomain 138.197.26.211"
11/29/2016 9:51:30 AMtime="2016-11-29T17:51:30Z" level=info msg="Adding Host k8svxlan3.localdomain 138.197.18.4"
11/29/2016 9:52:26 AMtime="2016-11-29T17:52:26Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:52:40 AMtime="2016-11-29T17:52:40Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:54:10 AMtime="2016-11-29T17:54:10Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:54:15 AMtime="2016-11-29T17:54:15Z" level=info msg="Adding Host k8svxlan1.localdomain 138.197.26.145"
11/29/2016 9:54:15 AMtime="2016-11-29T17:54:15Z" level=info msg="Adding Host k8svxlan2.localdomain 138.197.26.211"
11/29/2016 9:54:15 AMtime="2016-11-29T17:54:15Z" level=info msg="Adding Host k8svxlan3.localdomain 138.197.18.4"
11/29/2016 9:54:26 AMtime="2016-11-29T17:54:26Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:54:48 AMtime="2016-11-29T17:54:48Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:54:54 AMtime="2016-11-29T17:54:54Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:54:54 AMtime="2016-11-29T17:54:54Z" level=info msg="Adding Host k8svxlan1.localdomain 138.197.26.145"
11/29/2016 9:54:54 AMtime="2016-11-29T17:54:54Z" level=info msg="Adding Host k8svxlan2.localdomain 138.197.26.211"
11/29/2016 9:54:54 AMtime="2016-11-29T17:54:54Z" level=info msg="Adding Host k8svxlan3.localdomain 138.197.18.4"
11/29/2016 9:55:10 AMtime="2016-11-29T17:55:10Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:55:10 AMtime="2016-11-29T17:55:10Z" level=info msg="Adding Host k8svxlan1.localdomain 138.197.26.145"
11/29/2016 9:55:10 AMtime="2016-11-29T17:55:10Z" level=info msg="Adding Host k8svxlan2.localdomain 138.197.26.211"
11/29/2016 9:55:10 AMtime="2016-11-29T17:55:10Z" level=info msg="Adding Host k8svxlan3.localdomain 138.197.18.4"
11/29/2016 9:55:18 AMtime="2016-11-29T17:55:18Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:55:54 AMtime="2016-11-29T17:55:54Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:56:03 AMtime="2016-11-29T17:56:03Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:56:24 AMtime="2016-11-29T17:56:24Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:56:55 AMtime="2016-11-29T17:56:55Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:57:55 AMtime="2016-11-29T17:57:55Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:58:10 AMtime="2016-11-29T17:58:10Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:58:33 AMtime="2016-11-29T17:58:33Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:59:26 AMtime="2016-11-29T17:59:26Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:59:40 AMtime="2016-11-29T17:59:40Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 9:59:40 AMtime="2016-11-29T17:59:40Z" level=info msg="Adding Host k8svxlan1.localdomain 138.197.26.145"
11/29/2016 9:59:40 AMtime="2016-11-29T17:59:40Z" level=info msg="Adding Host k8svxlan2.localdomain 138.197.26.211"
11/29/2016 9:59:40 AMtime="2016-11-29T17:59:40Z" level=info msg="Adding Host k8svxlan3.localdomain 138.197.18.4"
11/29/2016 9:59:48 AMtime="2016-11-29T17:59:48Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host"
11/29/2016 10:00:10 AMtime="2016-11-29T18:00:10Z" level=error msg="Error getting IP addresses of host 640aac5c3227, err: lookup 640aac5c3227 on 169.254.169.250:53: no such host
```

**Comment by leodotcloud** [comment]
**Root cause:**
Container with IP: 10.42.251.203 started on host 2 and by the time the vxlan reload was happening, the vxlan container information was not found (10.42.84.207). The programming of the route failed but this was incorrectly cached. Hence the subsequent reloads didn't fix the problem.

**Possible fix:**
Don't populate the cache if the peer was not found.

```
time="2016-11-29T21:45:11Z" level=debug msg="Getting Entry from Container: {Name:healthcheck-healthcheck-3 PrimaryIp:10.42.251.203 PrimaryMacAddress:02:eb:78:6d:48:ad Ips:[10.42.251.203] Ports:[] ServiceName:healthcheck StackName:healthcheck Labels:map[io.rancher.stack.name:healthcheck io.rancher.container.create_agent:true io.rancher.project.name:healthcheck io.rancher.project_service.name:healthcheck/healthcheck io.rancher.service.hash:7896e52f6de45c242fa91bfefd65d056aa8fbad9 io.rancher.service.requested.host.id:14 io.rancher.scheduler.global:true io.rancher.service.deployment.unit:e1e2e948-eae1-499e-a14c-6145977da5b8 io.rancher.service.launch.config:io.rancher.service.primary.launch.config io.rancher.stack_service.name:healthcheck/healthcheck] CreateIndex:3 HostUUID:09edf6b4-3e74-4007-ae0c-38cd4af80e22 UUID:3d28dff1-5e0d-4515-ba18-69c13bcac479 State:starting HealthState: ExternalId: StartCount:0 MemoryReservation:0 MilliCPUReservation:0 Dns:[169.254.169.250] DnsSearch:[healthcheck.rancher.internal healthcheck.healthcheck.rancher.internal rancher.internal] HealthCheckHosts:[] NetworkFromContainerUUID: NetworkUUID:e409a086-7062-4688-84c2-7fadceac1365}"
time="2016-11-29T21:45:11Z" level=debug msg="entry: {IpAddress:10.42.251.203/16 HostIpAddress:45.55.83.227 Self:false Peer:false}"
time="2016-11-29T21:45:11Z" level=debug msg="entries: [{10.42.1.179/16 159.203.130.164 false false} {10.42.233.253/16 159.203.130.164 false false} {10.42.55.35/16 159.203.130.164 false false} {10.42.200.147/16 159.203.130.164 true true} {10.42.216.237/16 159.203.130.164 false false} {10.42.123.58/16 45.55.83.227 false false} {10.42.82.84/16 159.203.130.164 false false} {10.42.113.122/16 138.197.23.41 false false} {10.42.164.129/16 159.203.130.164 false false} {10.42.141.215/16 45.55.83.227 false false} {10.42.21.17/16 138.197.23.41 false false} {10.42.251.203/16 45.55.83.227 false false}]"
time="2016-11-29T21:45:11Z" level=debug msg="peersMap: map[10.42.200.147:{10.42.200.147/16 159.203.130.164 true true}]"
time="2016-11-29T21:45:11Z" level=info msg="vxlan: configure"
time="2016-11-29T21:45:11Z" level=debug msg="vxlan: checkAndCreateVTEP"
time="2016-11-29T21:45:11Z" level=debug msg="entries: map[10.42.200.147:{10.42.200.147/16 159.203.130.164 true true}]"
time="2016-11-29T21:45:11Z" level=debug msg="vxlan: buildPeersMapping"
time="2016-11-29T21:45:11Z" level=debug msg="entries: map[10.42.200.147:{10.42.200.147/16 159.203.130.164 true true}]"
time="2016-11-29T21:45:11Z" level=debug msg="peersMapping: map[string]net.IP{}"
time="2016-11-29T21:45:11Z" level=debug msg="newMap:map[10.42.200.147:{10.42.200.147/16 159.203.130.164 true true}]"
time="2016-11-29T21:45:11Z" level=debug msg="oldMap: map[10.42.200.147:{10.42.200.147/16 159.203.130.164 true true}]"
time="2016-11-29T21:45:11Z" level=debug msg="toAdd: map[]"
time="2016-11-29T21:45:11Z" level=debug msg="toDel: map[]"
time="2016-11-29T21:45:11Z" level=debug msg="toUpd: map[]"
time="2016-11-29T21:45:11Z" level=debug msg="newMap:map[10.42.123.58:{10.42.123.58/16 45.55.83.227 false false} 10.42.113.122:{10.42.113.122/16 138.197.23.41 false false} 10.42.141.215:{10.42.141.215/16 45.55.83.227 false false} 10.42.21.17:{10.42.21.17/16 138.197.23.41 false false} 10.42.251.203:{10.42.251.203/16 45.55.83.227 false false}]"
time="2016-11-29T21:45:11Z" level=debug msg="oldMap: map[10.42.123.58:{10.42.123.58/16 45.55.83.227 false false} 10.42.113.122:{10.42.113.122/16 138.197.23.41 false false} 10.42.141.215:{10.42.141.215/16 45.55.83.227 false false}]"
time="2016-11-29T21:45:11Z" level=debug msg="toAdd: map[10.42.21.17:{10.42.21.17/16 138.197.23.41 false false} 10.42.251.203:{10.42.251.203/16 45.55.83.227 false false}]"
time="2016-11-29T21:45:11Z" level=debug msg="toDel: map[]"
time="2016-11-29T21:45:11Z" level=debug msg="toUpd: map[]"
time="2016-11-29T21:45:11Z" level=debug msg="building remoteVxlanEntry for {10.42.21.17/16 138.197.23.41 false false}"
time="2016-11-29T21:45:11Z" level=debug msg="vxlan: Adding non peer remote : {ip:0xc420442bd0 via:[]}"
time="2016-11-29T21:45:11Z" level=debug msg="vxlan: adding route: 10.42.21.17/32 via <nil>"
time="2016-11-29T21:45:11Z" level=error msg="vxlan: error adding route: no such device"
time="2016-11-29T21:45:11Z" level=debug msg="building remoteVxlanEntry for {10.42.251.203/16 45.55.83.227 false false}"
time="2016-11-29T21:45:11Z" level=debug msg="vxlan: Adding non peer remote : {ip:0xc420442ed0 via:[]}"
time="2016-11-29T21:45:11Z" level=debug msg="vxlan: adding route: 10.42.251.203/32 via <nil>"
time="2016-11-29T21:45:11Z" level=error msg="vxlan: error adding route: no such device"
time="2016-11-29T21:45:11Z" level=debug msg="Reload successful"
time="2016-11-29T21:45:13Z" level=debug msg="Arp request for 10.42.0.1"
time="2016-11-29T21:45:14Z" level=debug msg="Metadata Version has been changed. Old version: 117-6c1101eb5d90330702b9e780b671ffc5. New version: 119-6c1101eb5d90330702b9e780b671ffc5."
time="2016-11-29T21:45:14Z" level=info msg="Metadata OnChange received, version: 119-6c1101eb5d90330702b9e780b671ffc5"
time="2016-11-29T21:45:14Z" level=info msg="vxlan: Reload"
```

**Comment by galal-hussein** [comment]
Rancher server: 1.5.0-alpha1

steps to validate:

- Enable VXLAN and disable ipsec
- Install Kubernetes

Creating and removing pods/rc/services etc. worked correctly and the stack was up and running without any problems.

**Comment by galal-hussein** [comment]
@leodotcloud When adding multiple nodes, kubernetes can only see one node.

**Comment by leodotcloud** [comment]
Fixed in `rancher/net:v0.9.3`

**Comment by galal-hussein** [comment]
Rancher server: master build

Steps to validate:

- Run Rancher server master build
- Add 3 hosts
- Install vxlan instead of ipsec on cattle
- Install k8s system stack
- Run `kubectl get nodes`

I can all 3 nodes when issuing the command, also i was able to run automation testing on it and the pods/rcs/services are up and running without issues
